src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Control/AsyncType.fs

```diff
@@ -147,8 +147,39 @@ type AsyncType() =
             Async.StartAsTask a
         this.WaitASec t
         Assert.IsTrue (t.IsCompleted)
-        Assert.AreEqual(s, t.Result)    
+        Assert.AreEqual(s, t.Result)
+
+    [<Test>]
+    member this.StartAsTaskCancellation () =
+        let cts = new CancellationTokenSource()
+        let tcs = TaskCompletionSource<unit>()
+        let a = async {
+            cts.CancelAfter (100)
+            do! tcs.Task |> Async.AwaitTask }
```

Comment:
It makes me a bit nervous that I don't see a way to remove this race condition (ie calling `cts.Cancel` after the bind)